src/ui/pages/slack_alerts_setup/form.jsx

```diff
@@ -12,73 +15,166 @@ import { ButtonGroup, FormWrapper } from './styles';
 export function SetupForm() {
     const [{ loading, error, data, isDirty }, update, save] = useSlackConfig();
 
+    const updateOauthToken = useCallback((e, { value }) => update({ ...data, slack_app_oauth_token: value }));
     const updateUrl = useCallback((e, { value }) => update({ ...data, webhook_url: value }));
     const updateUser = useCallback((e, { value }) => update({ ...data, from_user: value }));
     const updateUserIcon = useCallback((e, { value }) => update({ ...data, from_user_icon: value }));
 
+    const slackAppOauthToken = loading ? '' : data.slack_app_oauth_token;
     const webhookUrl = loading ? '' : data.webhook_url;
     const fromUserName = loading ? '' : data.from_user;
     const fromUserIcon = loading ? '' : data.from_user_icon;
 
+    const oauth_app_manifest = `
+display_information:
+  name: Splunk Alerts
+features:
+  bot_user:
+    display_name: Splunk
+    always_online: false
+oauth_config:
+  scopes:
+    bot:
+      - chat:write
+      - chat:write.public
+      - chat:write.customize
+settings:
+  org_deploy_enabled: false
+  socket_mode_enabled: false
+  token_rotation_enabled: false`.trim();
+
     return (
         <>
-            <Heading level={3}>Slack Incoming Webhook</Heading>
-            <Paragraph>
-                This alert action uses Slack's{' '}
-                <Link to="https://slack.com/apps/A0F7XDUAZ-incoming-webhooks" openInNewContext>
-                    Incoming Webhooks
-                </Link>{' '}
-                to post messages from Splunk into Slack channels. You can set a default webhook URL here,
-                which will be used for all alerts be default. Each alert can override and use a different
-                webhook URL that has different permissions or can send to a different Slack workspace.
-            </Paragraph>
-            <FormWrapper>
-                <ControlGroup
-                    label="Webhook URL"
-                    help={
+            <TabLayout defaultActivePanelId="oauth">
```

Comment:
These tabs are a little wordy, so definitely happy to take any wording suggestions you have. Overall, I think this came out well. Here's a snapshot:

![Screenshot from 2022-07-24 22-12-27](https://user-images.githubusercontent.com/2120379/180681057-2bbb7e04-d6a2-4bd7-81ea-dc35d1dd21e7.png)